johnzon-osgi/src/main/java/org/apache/johnzon/osgi/Config.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.johnzon.osgi;
+
+import javax.ws.rs.core.MediaType;
+
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+@ObjectClassDefinition(
+    description = "Configuration for Johnzon JSONB JAX-RS Whiteboard Message Body Reader & Writer",
+    name = "Johnzon JAX-RS JSONB",
+    pid = Config.CONFIG_PID
+)
+@interface Config {
```

Comment:
public? (to avoid unsafe requirement and some surprises in some env, not strictly required but saner IMHO)

Also do we want all the provider config there?

Reply:
I added more of the config options. Some you'd have to define an encoding for the `Map<String,String>` input since you can't do that on an annotation.